### ISSUE TITLE
shutil.get_terminal_size() returns inaccurate results

### ISSUE THREAD

**Issue by controversial** [omz/Pythonista-Issues#123]

**Comment by dgelessus** [comment]
What would you expect it to return? The console allows multiple fonts and sizes at once, you can't really say that the console is X by Y characters.

**Comment by controversial** [comment]
I'd expect it to return a value representing the maximum dimensions that `print` could put on the screen given the current settings.

**Comment by controversial** [comment]
It's typically used in terminal-based applications to know how wide they can print their contents to the screen. So it should cater to that functionality.

**Comment by lukaskollmer** [comment]
@controversial @dgelessus is right. It would only make sense for this function to return a character count, but that's not really possible since you can't know the font type or font size in advance. @omz could make it return the size of the output area in points, but there's no real use for such information because you still can't get a reliable character count from that.

**Comment by lukaskollmer** [comment]
One option would be to use the default output font to calculate line width but that wouldn't solve the number of lines problem since a previous output could have used a different font

**Comment by controversial** [comment]
@lukaskollmer Previous output doesn't need to be considered at all. The function just needs to return the amount of space that a normal `print` statement could take. The functionality of the function shouldn't change based on the environment. Just disregard `print` statements with previous fonts, pretend everything is done with `print`. `shutil` shouldn't contain Pythonista-specific functionality, it's a built-in.

**Comment by cclauss** [comment]
What happens on the Mac?!?

Thanks, CCC

**Comment by controversial** [comment]
It returns the width and height of the console, in characters. So,
something like 80x22 would indicate that you could fit 80 characters across
and 22 down. In other words, you could fill the screen by printing an
80-character string 22 times.

**Comment by cclauss** [comment]
Exactly, and when you change the font and/or the font size, the numbers change?

**Comment by controversial** [comment]
Yes. If you change the global settings, the numbers change. It still
doesn't make sense to account for previous print statements, what's
important is what you can fit with `print`

**Comment by jsbain** [comment]
it seems mainly one needs to know the columns.   rows is ambiguous in app, because of the keyboard.

**Comment by controversial** [comment]
I think rows should be implemented too, as the visible area. Think those command line interfaces that put up colorful unicode "GUI" interfaces on the screen. It should be what the user should see.

I don't really care much about rows myself, though.

**Comment by jsbain** [comment]
rows would just be confusing though ( you would get different results from the console vs running a script)

**Comment by controversial** [comment]
Yeah,  I guess. It should still try to be as accurate as possible, though, I think. Accuracy should be prioritized over consistency.

**Comment by Phuket2** [comment]
Guys, doesn't this older forum post cover a lot of this
https://forum.omz-software.com/topic/2554/how-can-i-get-console-width/28

**Comment by controversial** [comment]
@Phuket2 Yes, but this is a cross platform solution. The vast, vast, vast majority of Python libraries don't target Pythonista. This is something built into Python, it should work correctly. The question isn't whether it can be done, it's just that this should work in Pythonista and doesn't.

**Comment by Phuket2** [comment]
@controversial , fair enough. I really wonder if all the platforms do it exact. Maybe they do. But I would say technically this is a bit of a misnomer. The mean when does a console/terminal stop being one. I guess when they are bitmaps rather than ascii. Reporting a terminals rows/Cols should be std given we are talking about a fixed width font. But when dealing with proportional width/height chars, Unicode, emoji's etc, what's a poor developer to do 😱 Report back?
I am not taking sides, maybe the definition of terminals has changed over the years.
@controversial , I remember about talks of the console being slow sometimes. But if the terminal was pure ascii, I am sure it would be lightning fast. But having to deal with hyperlinks and every else know to man, it's probably one of the most complicated views in Pythonista.

Ok, just my 2 cents worth. Time to stagger home soon 😳😎

**Comment by dgelessus** [comment]
FWIW, in IDLE:

```python
Python 3.5.2 (v3.5.2:4def2a2901a5, Jun 26 2016, 10:47:25) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "copyright", "credits" or "license()" for more information.
>>> import shutil
>>> shutil.get_terminal_size()
os.terminal_size(columns=80, lines=24)
```

Which is just the default value. And IDLE's window layout is more like `xterm`/`Terminal.app`/`conhost.exe` than Pythonista. To be honest I would find it better to just leave it at the default, since it doesn't actually break scripts. If it bothers you, you can always set the `COLUMNS` and `LINES` envvars manually.

**Comment by controversial** [comment]
Yeah, but what's the downside to having it just work @dgelessus ? It would make Pythonista a lot more compatible with a lot of applications.

**Comment by Phuket2** [comment]
I am sure I will regret writing this, by diving into the deep end of the pool again, were I should not be swimming 😱
But wouldn't this be solved quite easily by a monkey_patch? 
But, I ask myself, is here just to report bugs request features etc or suggest work arounds.  It seems to me maybe it's a little off track. If something doesn't work and it should,miss a bug I guess. Not sure people are looking for work arounds here, just reporting on some facts.
Not trying to be harsh, I am doing the same here. But I question should I be offering a maybe work around. Seems like the only valid responses are to expand on an issue/bug/enhancement or point out that it's not a bug at all.

In a perfect world, I think you would have a link back to the forum to discuss the topic in my general terms. But for omz purposes, I guess he just needs to see a bug report.

It's ok, I am not going to heaven.

**Comment by jsbain** [comment]
>**shutil.get_terminal_size(fallback=(columns, lines))**
>Get the size of the terminal window.

>For each of the two dimensions, the environment variable, COLUMNS and LINES respectively, is checked. If the variable is defined and the value is a positive integer, it is used.

>When COLUMNS or LINES is not defined, which is the common case, the terminal connected to sys.__stdout__ is queried by invoking os.get_terminal_size().

>If the terminal size cannot be successfully queried, either because the system doesn’t support querying, or because we are not connected to a terminal, the value given in fallback parameter is used. fallback defaults to (80, 24) which is the default size used by many terminal emulators.

``` 
>>> os.environ['COLUMNS']='72'
>>> shutil.get_terminal_size()
os.terminal_size(columns=72, lines=24)
``` 
Per the standardlib docs, this behaves exactly as expected.  In this case, "the system does not suppor querying".   You can set the environ vars if you want to use a program that uses it.
This is not a bug -- it is an enhancement request.  The downside is we would like bugs fixed :D, though obviously omz is going to work on what he thinks is important or interesting, regardless of what labels applies.
